### PR TITLE
workflows/triage: more improvements

### PR DESCRIPTION
**Pull request by SeekingMeaning**
Continuation of https://github.com/Homebrew/homebrew-core/pull/69975

- `?:` is not needed since `Homebrew/actions/label-pull-requests` doesn't do anything with captures.
- Added `(@[0-9.]+)?` to all dependency regexes for consistency. Although some formulae such as `rust` don't have versioned formulae, it doesn't hurt to search for them anyway; this could useful if we ever add them in the future.
- Before labelling, sleep for 10 seconds. I'm not sure why, but it seems CI is not noticing the `automerge-skip` label when it is added during PR creation. In https://github.com/Homebrew/homebrew-core/pull/69994, the initial `triage` job on PR open doesn't list `automerge-skip` as one of the existing labels: https://github.com/Homebrew/homebrew-core/runs/1793465318?check_suite_focus=true#step:4:106 (Notably, `CI-syntax-only` and `do not merge` are listed, but not `automerge-skip`).

### CODE REVIEW

File: .github/workflows/triage.yml [hunk 1]
```diff
@@ -66,52 +69,52 @@ jobs:
...
+                    "content": "depends_on \"openjdk(@[0-9.]+)?\""
                 }, {
                     "label": "linux-only",
                     "path": "Formula/.+",
                     "content": "depends_on :linux"
                 }, {
                     "label": "lua",
                     "path": "Formula/.+",
-                    "content": "depends_on \"lua(?:@[0-9.]+|jit)?\""
+                    "content": "depends_on \"lua(jit)?(@[0-9.]+)?\""
```
Comment (carlocab) [inline .github/workflows/triage.yml:88]: Can you also include `luajit-openresty` here?

Comment (SeekingMeaning) [reply .github/workflows/triage.yml:88]: Oh yeah thanks for reminding me

**Review by carlocab — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by SeekingMeaning** [comment]
For reference: a regex broke because `\[` needed to be `\\[`; fixed in commit 6c76d724e995e209f9efe13cf50f9389d2afe15f

**Comment by Rylan12** [comment]
@SeekingMeaning the `automerge-skip` skip is added/removed during the `Check commit format` job which was before the sleep.

If it's not been resolved, I'll take a look and see if I can figure out what's going wrong but I don't have a ton of time.

**Comment by SeekingMeaning** [comment]
> @SeekingMeaning the `automerge-skip` skip is added/removed during the `Check commit format` job which was before the sleep.

I just figured that out, haha

The conclusion I reached: https://github.com/Homebrew/homebrew-core/pull/69996#issuecomment-770041128

> Edit: was the issue that the `automerge-skip` label was removed after _manually add it during creation_ or after _automatically being added after creation_?

`automerge-skip` was removed after manually adding it during creation, which appears to be the intended behavior

**Comment by Rylan12** [comment]
Okay, good. Yeah, it's intended because we want to un-mark PRs as `automerge-skip` if they're modified to meet the guidelines. So for a manual use case `do not merge`is  probably better,

**Pull request merged**